lib/bolt/catalog.rb

```diff
@@ -54,89 +56,97 @@ def generate_ast(code, filename = nil)
     end
 
     def compile_catalog(request)
-      pal_main = request['code_ast'] || request['code_string']
-      target = request['target']
       pdb_client = Bolt::PuppetDB::Client.new(Bolt::PuppetDB::Config.new(request['pdb_config']))
-      options = request['puppet_config'] || {}
       project = request['project'] || {}
       bolt_project = Struct.new(:name, :path).new(project['name'], project['path']) unless project.empty?
-      with_puppet_settings(request['hiera_config']) do
-        Puppet[:rich_data] = true
-        Puppet[:node_name_value] = target['name']
-        env_conf = { modulepath: request['modulepath'] || [],
-                     facts: target['facts'] || {} }
-        env_conf[:variables] = {}
+      inv = Bolt::ApplyInventory.new(request['config'])
+      puppet_overrides = {
+        bolt_pdb_client: pdb_client,
+        bolt_inventory: inv,
+        bolt_project: bolt_project
+      }
+
+      # Facts will be set by the catalog compiler, so we need to ensure
```

Comment:
I think this actually belongs above the definition of `shadow_vars`, since this describes exactly what that function is doing.

Reply:
Or to above `plan_vars`, whichever you think makes more sense.

Reply:
Or removed altogether, the current comment above the function definition does a pretty good job of getting the important point across

Reply:
Maybe just the `target = request['target']` makes that a bit awkward. We need that `target` var there to access facts. The description i think is meant to descrive _how_ and _why_ topscope_vars is computed.

Reply:
I just thought it might be more clear placed elsewhere, but I don't think it's worth holding this up for.